src/Parsers/ParserCreateFunctionQuery.cpp

```diff
@@ -110,6 +131,33 @@ bool ParserCreateFunctionQuery::parseImpl(IParser::Pos & pos, ASTPtr & node, Exp
         if (!lambda_p.parse(pos, function_core, expected))
             return false;
     }
+    Poco::JSON::Object::Ptr payload = new Poco::JSON::Object();
+    if (is_remote){
+        if (is_aggregation){
+            throw Exception("Remote udf can not be an aggregate function",ErrorCodes::AGGREGATE_FUNCTION_NOT_APPLICABLE);
+        }
+        if (!s_url.ignore(pos,expected))
+            return false;
+        if (!value.parse(pos, url, expected))
+            return false;
+        function_core = std::make_shared<ASTLiteral>(Field());
+        payload->set("URL", url->as<ASTLiteral>()->value.safeGet<String>());
+        if (s_auth_method.ignore(pos,expected)){
+            if (!value.parse(pos, auth_method, expected))
```

Comment:
the parsing logic is incorrect here. it should be like:
1. if auth_method == 'none' skip the parsing( no other auth settings left)
2. if auth_method =='auth_header', try to parse AUTH_KEY and AUTH_VALUE
3. if AUTH_KEY or AUTH_VALUE is missing, return false (parse fail)
4. if auth_method != 'none' and auth_method !='auth_header', throw Exception